app/src/main/java/org/mozilla/fenix/components/metrics/LeanplumMetricsService.kt

```diff
@@ -46,11 +46,31 @@ private val Event.name: String
 }
 
 class LeanplumMetricsService(private val application: Application) : MetricsService {
+    data class Token(val id: String, val token: String) {
+        enum class Type { Development, Production, Invalid }
+
+        val type by lazy {
+            when {
+                token.take(4) == "prod" -> Type.Production
+                token.take(3) == "dev" -> Type.Development
+                else -> Type.Invalid
+            }
+        }
+    }
+
+    private val token = Token(LeanplumId, LeanplumToken)
+
     override fun start() {
+        when(token.type) {
+            Token.Type.Production ->  Leanplum.setAppIdForProductionMode(token.id, token.token)
+            Token.Type.Development -> Leanplum.setAppIdForDevelopmentMode(token.id, token.token)
+            Token.Type.Invalid -> return
```

Comment:
Might be good to send something here anyway so you can tell if something is wrong rather than just dropping it on the floor. We had someone fork our app, who obviously didn't have the right keys, and we couldn't figure out who bc we didn't program defensively. Not sure you can do that here if the token has to do with endpoint though.